app/common/expressions/managed.py

```diff
@@ -116,15 +302,63 @@ def statement(self) -> str:
             f"{self.maximum_value}"
         )
 
+    @staticmethod
+    def get_form_fields() -> dict[str, "Field"]:
+        return {
+            "between_bottom_of_range": IntegerField(
+                "Minimum value",
+                widget=GovTextInput(),
+                validators=[Optional()],
+                render_kw={"params": {"classes": "govuk-input--width-10"}},
+            ),
+            "between_bottom_inclusive": BooleanField(
+                "An answer of exactly the minimum value is allowed", widget=GovCheckboxInput()
+            ),
+            "between_top_of_range": IntegerField(
+                "Maximum value",
+                widget=GovTextInput(),
+                validators=[Optional()],
+                render_kw={"params": {"classes": "govuk-input--width-10"}},
+            ),
+            "between_top_inclusive": BooleanField(
+                "An answer of exactly the maximum value is allowed", widget=GovCheckboxInput()
+            ),
+        }
+
+    @staticmethod
+    def update_validators(form: "_ManagedExpressionForm") -> None:
+        form.between_bottom_of_range.validators = [
+            InputRequired("Enter the minimum value allowed for this question"),
+            BottomOfRangeIsLower("The minimum value must be lower than the maximum value"),
+        ]
+        form.between_top_of_range.validators = [
+            InputRequired("Enter the maximum value allowed for this question"),
+            BottomOfRangeIsLower("The maximum value must be higher than the minimum value"),
+        ]
+
+    @staticmethod
+    def build_from_form(form: "_ManagedExpressionForm", question: "Question") -> "Between":
+        return Between(
+            question_id=question.id,
+            minimum_value=form.between_bottom_of_range.data,
+            minimum_inclusive=form.between_bottom_inclusive.data,
+            maximum_value=form.between_top_of_range.data,
+            maximum_inclusive=form.between_top_inclusive.data,
+        )
+
+    @staticmethod
+    def form_data_from_expression(expression: "Expression") -> dict[str, Any]:
+        return {
+            "between_bottom_of_range": expression.context["minimum_value"],
+            "between_bottom_inclusive": expression.context["minimum_inclusive"],
+            "between_top_of_range": expression.context["maximum_value"],
+            "between_top_inclusive": expression.context["maximum_inclusive"],
+        }
```

Comment:
I wonder if because the form is all already being built dynamically if this mapping could be rolled into `get_form_fields()` which would optionally accept an expression (if it already exists) and set the `default=` property if so

The one advantage to that is it would stop the string version of the form keys i.e "between_bottom_of_range" needing to be re-mapped and line up between where its defined three methods above and here

I'm sure there are some cons of not being able to get this mapping separately so will just note it down as an idea - all of the other methods _I think_ would break at runtime if they were trying to access keys on the form that didn't exist vs. this one might quite silently not work

Reply:
(I don't think this should be changed right now unless you v. strongly agree for some reason)